interbtc-spec/docs/source/spec/security.rst

```diff
@@ -124,53 +121,42 @@ Specification
 
 *Parameters*
 
-* ``account``: Parachain account identifier (links this identifier to the AccountId associated with the process where this secure id is to be used, e.g., the user calling :ref:`requestIssue`).
-
-*Returns*
+* ``account``: account identifier (links this identifier to the AccountId associated with the process where this secure id is to be used, e.g., the user calling :ref:`requestIssue`).
 
-* ``hash``: a cryptographic hash generated via a secure hash function.
+*Preconditions*
 
-Function Sequence
-.................
+* MUST combine ``account``, ``Nonce`` and the hash of the parent block.
 
-1. Increment the ``Nonce``.
-2. Concatenate ``account``, ``Nonce``, and ``parent_hash()``.
-3. SHA256 hash the result of step 1.
-4. Return the resulting hash.
+*Postconditions*
 
-.. note:: The funtion ``parent_hash()`` is assumed to return the hash of the parachain's parent block - which precedes the block this function is called in.
+* Nonce MUST be incremented by one.
+* MUST return the 256-bit hash of the input fields.
 
 
 .. _hasExpired:
 
 hasExpired
 ----------
 
-Checks if the given period has expired since the given starting point. This calculation is based on the :ref:`activeBlockCount`.
+Checks if the ``period`` has expired since the ``opentime``. This calculation is based on the :ref:`activeBlockCount`.
 
 Specification
 .............
 
 *Function Signature*
 
-``has_expired(opentime, period)``
+``hasExpired(opentime, period)``
 
 *Parameters*
 
 * ``opentime``: the :ref:`activeBlockCount` at the time the issue/redeem/replace was opened.
+* ``period``: the number of blocks the user or Vault has to complete the action.
 
-* ``period``: the number of blocks the user or vault has to complete the action.
-
-
-*Returns*
-
-* ``true`` if the period has expired
+*Preconditions*
```

Comment:
```suggestion
*Preconditions*

* The `ActiveBlockCount` MUST be greater than 0.
```

Note, strictly speaking, we could check that the open_time <= ActiveBlockCount, otherwise there is an error in the implementation.